job-descriptions/software-engineer-code-exercise.md

```diff
@@ -0,0 +1,60 @@
+# Code exercise
+
+We ask candidates to complete a 2-hour asynchronous coding exercise so we can evaluate you on certain skills that we think are relevant to being successful at Sourcegraph.
+
+Why? Read the [FAQ](#FAQ).
+
+## TypeScript code exercise
+
+We designed this exercise to measure your understanding of callbacks and asyncronous execution.
+
+- You will choose a two hour timeframe to independently work on the exercise. You can choose any day and time; we just schedule an email.
+- At your chosen date and time, we will send you a zipped TypeScript project that contains instructions and some unit tests.
+- TypeScript is a superset of JavaScript, so if you are comfortable with JavaScript and have experience reading any typed language, then you should be fine. We recommend reading through the [TypeScript docs](https://www.typescriptlang.org/docs/handbook/basic-types.html) if you aren't already familiar with TypeScript.
+- You should have Node.js (>=8.16.0) installed on your computer.
+- You can look up documentation on the internet while you are coding.
+- You may use open-source libraries, but most candidates don't find it necessary.
```

Comment:
I think in at least one case we ask candidates not to: Maybe "Unless instructed otherwise, …" ?

Reply:
@lguychard should we tell candidates to not use libraries here? Are there good candidates who use libraries?

Reply:
I updated the language, thanks!